### PR TITLE
[INLONG-397][Doc] Add guide for extend data source in manager

### LINKED ISSUE #397: [Doc] Add blog for extend data source in manager
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by kipshi**
Fixes #397 

where *XYZ* should be replaced by the actual issue number.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: docs/design_and_concept/how_to_extend_data_source.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
+---
+title: Extended Data Source
```
Comment (dockerzhang) [inline docs/design_and_concept/how_to_extend_data_source.md:2]: ->
Data Node Plugin

File: docs/design_and_concept/how_to_extend_data_source.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+sidebar_position: 6
+---
+
+## Overview
+
+Inlong is aimed at create dataflow between different data sources, now Inlong has support several universal data sources such as **MySQL**, **Apache Kafka**, **ClickHouse** on Input/Output respectively,
+You can refer to [data_node](https://inlong.apache.org/docs/next/data_node/extract_node/auto_push) for specific information.
+We Plan to support more data sources in the future, and this article is a development manual to extend data sources.
```
Comment (dockerzhang) [inline docs/design_and_concept/how_to_extend_data_source.md:10]: data sources
->
data nodes

File: docs/design_and_concept/how_to_extend_data_source.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+## Extend Data Extract Node
```
Comment (dockerzhang) [inline docs/design_and_concept/how_to_extend_data_source.md:12]: Extend Extract Node

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_data_source.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
+---
+title: 数据源扩展
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_data_source.md:2]: 数据节点插件

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_data_source.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+            case MONGODB:
+                return createExtractNode((MongoDBSource) sourceInfo);
+            default:
+                throw new IllegalArgumentException(
+                        String.format("Unsupported sourceType=%s to create extractNode", sourceType));
+        }
+    }
+    ...
+```
+## 扩展写入节点
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_data_source.md:81]: need one more line.

File: docs/design_and_concept/how_to_extend_data_source.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+            case MONGODB:
+                return createExtractNode((MongoDBSource) sourceInfo);
+            default:
+                throw new IllegalArgumentException(
+                        String.format("Unsupported sourceType=%s to create extractNode", sourceType));
+        }
+    }
+    ...
+```
+## Extend Data Load Node
```
Comment (dockerzhang) [inline docs/design_and_concept/how_to_extend_data_source.md:81]: need one more line.

**Comment by dockerzhang** [comment]
how_to_extend_data_source.md
->
inlong_data_node_plugin.md

**Review by healchow — APPROVED** [review]

**Pull request merged**